### ISSUE TITLE
Field attributes on derive macros produces a false positive error message

### ISSUE THREAD

**Issue by acycliczebra** [rust-lang/rust-analyzer#18224]
**rust-analyzer version**: 0.3.2129

**rustc version**: 1.80.1

**editor or extension**: VSCode

**relevant settings**: None

**repository link (if public, optional)**: https://github.com/acycliczebra/rust-analyzer-derive-issue

**code snippet to reproduce**: 

Please see repo for the full example.

I ran into this issue when trying to create a derived struct with different fields than the original. Rust analyzer gives an error regarding missing fields, even though the program compiles fine.

![Image](https://github.com/user-attachments/assets/5d824539-24f1-4bb8-a6a4-8808a439b85f)

Oddly enough, this issue only exists for field attributes and not for the top level derive input attributes. I temporarily worked around this by changing my macro to accept something like this instead (which is not ideal):

```rust

#[derive(WithIgnoredFields)]
#[ignore_field(field_b)]
struct MyStruct {
    field_a: &'static str,
    field_b: &'static str,
}

```

**Comment by ChayimFriedman2** [comment]
This comes from difference stringification of `TokenStream`. The attribute stringifies to `# [ignore_field]` (notice the space). Given that [the exact way a token stream is stringified is unspecified and the compiler is allowed to change it](https://doc.rust-lang.org/nightly/std/macro.stringify.html), I consider this a non-bug.
